File: services/dotnet-svc/Models/Payment.cs
namespace DotnetSvc.Models;

public sealed class Payment
{
    public long Id { get; set; }
    public long OrderId { get; set; }
    public long CustomerId { get; set; }
    public long AmountCents { get; set; }
    public string Status { get; set; } = "AUTHORIZED";
    public DateTime CreatedAt { get; set; }
}

File: services/helidon-mp-svc/src/main/java/com/perfsim/helidonmpsvc/domain/Payment.java
package com.perfsim.helidonmpsvc.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.OffsetDateTime;

@Entity
@Table(name = "payments", schema = "helidon_mp")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(name = "order_id", nullable = false)
    public Long orderId;

    @Column(name = "customer_id", nullable = false)
    public Long customerId;

    @Column(name = "amount_cents", nullable = false)
    public long amountCents;

    @Column(nullable = false, length = 32)
    public String status = "AUTHORIZED";

    @Column(name = "created_at", nullable = false, insertable = false, updatable = false)
    public OffsetDateTime createdAt;
}
